sys/shell/shell.c

```diff
@@ -44,6 +44,12 @@ static void _putchar(int c) {
 #endif
 #endif
 
+#ifdef MODULE_NEWLIB
+#define flush_if_needed()   (fflush(stdout))
```

Comment:
Please don't define a macro that looks like a function.
I'd prefer a real function with the ifdef inside:

```
static void flush_if_needed(void)
{
#ifdef MODULE_NEWLIB
...
#endif
}
```